lib/utils/template_generator.js

```diff
@@ -74,79 +79,37 @@ class TemplateGenerator {
-
-        sadly it doesn't extract from http(s)://github.com/status-im/dappcon-workshop-dapp/tree/start-here
-        thats why we have a special case later
-    */
-    const match = uri.match(/github\.com+\/(.+?)(#.*)?$/);
-    const githubPart = "https://github.com/";
-    let repoPart = match !== null? match[1] : null;
-    let branchName = match !== null? match[2] : null;
-
-    if (branchName && branchName !== '#'){
-      branchName = branchName.substring(1);
-    } else {
-      branchName = "master";
+      utils.runCmd('npm install', null, (err) => {
+        if (err) {
+          return console.error(err.red);
+        }
+        console.log(__('Init complete').green);
+        console.log('\n' + __('App ready at ').green + templatePath);
+      });
     }
-
-    let url, folder;
-    if (uri.includes("/tree")){
-    //e.g http(s)://github.com/status-im/dappcon-workshop-dapp/tree/start-here
-      let repoPartAndBranch = repoPart.split("/tree/");
-      repoPart = repoPartAndBranch[0];
-      branchName = repoPartAndBranch[1];
-      url = "https://github.com/" + repoPart + "/archive/"+ branchName +".zip";
-      folder = repoPart + "/" + branchName;
-    } else if (repoPart !== undefined) {
-      url = githubPart + repoPart + "/archive/" + branchName + ".zip";
-      folder = repoPart + "/" + branchName;
-   }
-
-    return {
-      'url': url,
-      'folder': folder
-    };
   }
 
   getExternalProject(uri) {
-    let url, folder;
-    const regex = /^((git@)?(www\.)?github\.com\/)|(https?:\/\/)/;
-    if (!uri.match(regex) && uri.split('/').length >= 2) {
-      //e.g embark-framework/embark, embark-framework/embark#branch, embark-framework/embark#features/branch
-      let repoPartAndBranch = uri.split('#');
-      let repoPart = repoPartAndBranch[0];
-      let branchName = (repoPartAndBranch.length === 2)? repoPartAndBranch[1] : "master";
-      url = "https://github.com/" + repoPart + "/archive/"+ branchName + ".zip";
-      folder = repoPart + "/" + branchName;
-    } else if (uri.indexOf('/') === -1) {
-      url = "https://github.com/embark-framework/embark-" + uri + "-template/archive/master.zip";
-      folder = "embark-framework/embark-" + uri + "-template";
-    } else {
-      let urlAndFolder = this.extractGithubUrlAndFolder(uri);
-      url = urlAndFolder.url;
-      folder = urlAndFolder.folder;
+    let url, folder, hgi;
+    try {
+      hgi = hostedGitInfo.fromUrl(uri);
+      if (!hgi || hgi.user.includes('#')) {
+        let templateAndBranch = uri.split('#');
+        templateAndBranch[0] = `embark-framework/embark-${templateAndBranch[0]}-template`;
+        hgi = hostedGitInfo.fromUrl(templateAndBranch.join('#'));
+      }
+      if(!hgi) { throw new Error(); }
+      url = hgi.tarball();
+      folder = `${hgi.user}/${hgi.project}/${hgi.committish || 'master'}`;
+    } catch (e) {
+      console.error('Unsupported template name or git host URL'.red);
```

Comment:
Not a deal breaker, just a thought: I wonder if this error message is better placed inside `new Error()`  within the `try` block and then we'd just `console.error(e)` here.

Reply:
Definitely an option. By placing it inside `catch (e) {...}`, I intended it as a catch-all error message, i.e. avoiding exposure of potentially cryptic error messages (if any) originating from `hosted-git-info`.

Reply:
Right. I don't know what other errors `hosted-git-info` might expose. Could be confusing though if some other error is emitted, but we tell the user that it's because of an unsupported template name or host URL (even though it might be something totally different).